src/py42/modules/archive.py

```diff
@@ -1,33 +1,100 @@
 class ArchiveModule(object):
+    """A module for getting information about archives on storage nodes along with functionality
+    for streaming a file from backup.
+    """
+
     def __init__(self, archive_accessor_manager, archive_client):
         self._archive_accessor_manager = archive_accessor_manager
         self._archive_client = archive_client
 
-    def stream_from_backup(self, file_path, device_guid, destination_guid=None):
+    def stream_from_backup(
+        self,
+        file_path,
+        device_guid,
+        destination_guid=None,
+        archive_password=None,
+        encryption_key=None,
+    ):
+        """Streams a file from an archive to memory.
+        `REST Documentation <https://console.us.code42.com/apidocviewer/#WebRestoreJobResult-get>`__
+
+        Args:
+            file_path (str): The path to the file in your archive.
+            device_guid (str): The GUID for the device the file belongs to.
+            destination_guid (str, optional): The GUID for the server that stores the backup of
+                the file. If None, it will use the first destination GUID it finds for your
+                device. 'destination_guid' may be useful if the file is missing from one of your
+                destinations or if you want to optimize performance. Defaults to None.
+            archive_password (str, None): The password for archives that are protected with an
+                additional password. This is only relevant to users with this level of archive
+                    security. Defaults to None.
+            encryption_key (str, None): A custom encryption key for decryption an archive's file
+                contents, necessary for restoring files. This is only relevant to users with
+                this level of archive security. Defaults to None.
+        Returns:
+            :class:`py42.sdk.response.Py42Response`: A response containing the streamed content.
```

Comment:
We should probably add some description of how to handle the streamed data and get it into a usable file on disk.

Reply:
Agreed - I'll add a usage block